### PR TITLE
docs: add migration guide for v4

### PR DESCRIPTION
**Pull request by dpopp07**
Signed-off-by: Dustin Popp <dpopp07@gmail.com>

### CODE REVIEW

File: MIGRATION-V4.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Code produced by the IBM OpenAPI SDK Generator should already be compliant with TypeScript v4.
+
+Additionally, you may need to upgrade `@typescript-eslint/eslint-plugin` to be compatible with the new
+version of TypeScript. This upgrade should be safe and compatible, especially if you remain on major version 4
+of this development dependency.
+
+Run `npm i -D '@typescript-eslint/eslint-plugin'` to upgrade.
+
+### Jest Version
+The `jest` development dependnecy is now required to be at least version 29.
```
Comment (padamstx) [inline MIGRATION-V4.md:22]:
```suggestion
The `jest` development dependency is now required to be at least version 29.
```

**Review by padamstx — APPROVED** [review]
Looks good except for one little typo :)
I'll approve now though

File: MIGRATION-V4.md (hunk 1 above)
Comment (dpopp07) [reply MIGRATION-V4.md]: Good catch!

**Pull request merged**